source/src/main/java/com/adataptivescale/rosetta/source/core/extractors/table/KineticaTablesExtractor.java

```diff
@@ -20,28 +26,57 @@ public class KineticaTablesExtractor extends DefaultTablesExtractor {
     @Override
     public Collection<Table> extract(Connection target, java.sql.Connection connection) throws SQLException {
         Collection<Table> tables = super.extract(target, connection);
-
-        return attachTableType(tables, connection);
+        attachTablePartitions(tables, connection);
+        attachTableType(tables, connection);
+        return tables;
     }
 
-    private Collection<Table> attachTableType(Collection<Table> tables, java.sql.Connection connection) throws SQLException {
+    private void attachTableType(Collection<Table> tables, java.sql.Connection connection) throws SQLException {
         ResultSet resultSet = connection.createStatement().executeQuery("SELECT object_name, schema_name, shard_kind, persistence FROM ki_catalog.ki_objects;");
 
         while (resultSet.next()) {
             String object_schema = resultSet.getString("schema_name");
             String object_name = resultSet.getString("object_name");
             Optional<Table> found_table = tables.stream().filter(table -> table.getSchema().equals(object_schema) && table.getName().equals(object_name)).findFirst();
             if (found_table.isPresent()) {
-                Map<String, Object> additionalProps = new HashMap<>();
-                additionalProps.put("shard_kind", resultSet.getString("shard_kind"));
-                additionalProps.put("persistence", resultSet.getString("persistence"));
-                found_table.get().setAdditionalProperties(additionalProps);
+                found_table.get().addProperty("shard_kind", resultSet.getString("shard_kind"));
+                found_table.get().addProperty("persistence", resultSet.getString("persistence"));
             }
         }
 
         if (!resultSet.isClosed()) {
             resultSet.close();
         }
-        return tables;
+    }
+
+    private void attachTablePartitions(Collection<Table> tables, java.sql.Connection connection) {
+        for (Table table : tables) {
+            String queryTemplate = "SHOW %s.%s;";
+            try (Statement statement = connection.createStatement()) {
+                String query = String.format(queryTemplate, table.getSchema(), table.getName());
+                ResultSet resultSet = statement.executeQuery(query);
+                List<Map<String, Object>> records = QueryHelper.mapRecords(resultSet);
+
+                if (!records.isEmpty()) {
+                    table.addProperty("partitions", extractPartitions(records.get(0).get("ddl").toString()));
+                }
+
+            } catch (SQLException e) {
+                log.warn("Skipping processing views due to error: {}", e.getMessage());
```

Comment:
Do we need to say here `Skipping processing views ...`

Reply:
No, I forgot to modify the warn message, now it should be good.